files/build_templates/buffers_config.j2

```diff
@@ -182,6 +190,8 @@ def
 {{ defs.generate_queue_buffers(port_names_active) }}
 {% elif defs.generate_queue_buffers_with_inactive_ports is defined %}
 {{ defs.generate_queue_buffers_with_inactive_ports(port_names_active, port_names_inactive) }}
+{% elif generate_extra_lossless_buffers is defined %}
```

Comment:
1. Looks like the name `generate_extra_lossless_buffers` is a bit misleading.
   The macro is introduced to generate queue buffers for both lossless and lossy queues with additional lossless queues on ports between leaf routers and dual ToR routers. Maybe `generate_queue_buffers_with_extra_lossless_queues` or `generate_queue_buffers_dual_tor` is better.
2. Some vendors, like Nvidia, provide a `_with_inactive_ports` version queue/PG buffer generating macros for the purpose of reclaiming unused buffer. In this version, an extra argument `port_names_inactive` is passed.
So you may want to provide a `_with_inactive_ports` version for generating queue buffers for ports connecting to dual ToR.
3. The macro is defined in per-SKU `buffer_config_tx.j2` which was imported as `defs`. So we need `defs.` to reference it, like other macros

Reply:
Thanks for the grest suggetsions.
For 1, I renamed the macro to `generate_queue_buffers_with_extra_lossless_queues`. It's more accurate. Thanks
For 2, I added the call of macro `generate_queue_buffers_with_extra_lossless_queues_with_inactive_ports` in `buffers_config.j2` . But the macro is not defined yet. We may implement the macro when we are updating template for NVIDIA devices
For 3, I didn't do the change as I think it makes more sense to define the macro `generate_queue_buffers_with_extra_lossless_queues` in `buffers.json.j2` as the macro is exactly the same for different SKUs and roles (T0 and T1). If we move the macro into `buffers_defaults_t0.j2` or `buffers_defaults_t1.j2`, that means lots of copy-paste. 
BTW, I also update the macro to accept port names joint with `,` like the existing macros.

Thanks

Reply:
Thanks. I added a `refs.` in front the call of `generate_queue_buffers_with_extra_lossless_queues_and_inactive_ports` so that vendors can define this macros as other marcos.